backend/src/main/kotlin/blue/mild/covid/vaxx/service/IsinRetryService.kt

```diff
@@ -0,0 +1,153 @@
+    suspend fun runIsinRetry(isinJobDto: IsinJobDtoIn): IsinJobDtoOut {
+        val stats = IsinJobDtoOut()
+
+        val patients = patientService.getPatientsByConjunctionOf(
+            n = isinJobDto.patientsCount,
+            offset = isinJobDto.patientsOffset.toLong()
+        )
+
+        for (patient in patients) {
+            logger.debug("Checking ISIN id of patient ${patient.id}")
+
+            // 1. If patient has personal number but ISIN id is not set -> try ISIN validation
+            val updatedPatient = if (!patient.isinId.isNullOrBlank()) {
+                patient
+            } else if (isinJobDto.validatePatients && !patient.personalNumber.isNullOrBlank() ) {
+                logger.info("Patient ${patient.id} has personal number but no ISIN id. Validating in ISIN...")
+
+                val newIsinPatientId = retryPatientValidation(patient)
+
+                if (newIsinPatientId != null) {
+                    stats.validatedPatientsSuccess++;
+                } else {
+                    stats.validatedPatientsErrors++;
+                }
+                patient.copy(isinId = newIsinPatientId)
+            } else {
+                patient.copy(isinId = null)
+            }
+
+            if (updatedPatient.isinId.isNullOrBlank()) continue
+
+            // 2. If data are correct but not exported to ISIN -> try export to isin
+            logger.debug("Checking correctness exported to ISIN of patient ${updatedPatient.id}")
+            if (
+                isinJobDto.exportPatientsInfo &&
+                updatedPatient.dataCorrect != null &&
+                updatedPatient.dataCorrect.dataAreCorrect &&
+                updatedPatient.dataCorrect.exportedToIsinOn == null
+            ) {
+                val wasExported = retryPatientContactInfoExport(updatedPatient)
+
+                if (wasExported) {
+                    stats.exportedPatientsInfoSuccess++
+                } else {
+                    stats.exportedPatientsInfoErrors++
+                }
+            }
+
+            // 3. If vaccinated but not vaccination is not exported to ISIN -> try export vaccination to isin
+            if (isinJobDto.exportVaccinations && updatedPatient.vaccinated != null && updatedPatient.vaccinated.exportedToIsinOn == null) {
+                val wasExported = retryPatientVaccinationCreation(updatedPatient)
+
+                if (wasExported) {
+                    stats.exportedVaccinationsSuccess++
+                } else {
+                    stats.exportedVaccinationsErrors++
+                }
+            }
+        }
+
+        return stats
+    }
+
+    private suspend fun retryPatientValidation(patient: PatientDtoOut): String? {
+        val patientValidationResult = patientValidation.validatePatient(
+            firstName = patient.firstName,
+            lastName = patient.lastName,
+            personalNumber = patient.personalNumber ?: throw AssertionError { "Personal number cannot be null."},
```

Comment:
`requireNotNull(patient.personalNumber) { "Personal number for patient ${patient.id} can not be null." }`

Reply:
Tohle v dalším PR s validací cizinců předělávám